sdk/identity/azure-identity/azure/identity/aio/_credentials/cli_credential.py

```diff
@@ -0,0 +1,58 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import json
+import time
+from datetime import datetime
+
+from subprocess import Popen, PIPE
+import six
+
+from azure.core.credentials import AccessToken
+from azure.core.exceptions import ClientAuthenticationError
+
+
+class AzureCliCredential(object):
+
+    _DEFAULT_PREFIX = "/.default"
+    _CLI_NOT_INSTALLED_ERR = "Azure CLI not installed"
+    _CLI_LOGIN_ERR = "ERROR: Please run 'az login' to setup account.\r\n"
+
+    async def get_token(self, *scopes, **kwargs): # pylint:disable=unused-argument
+        command = 'az account get-access-token'
+        
+        if scopes:
+            resource = scopes[0]
+            if resource.endswith(self._DEFAULT_PREFIX):
+                resource = resource[:-len(self._DEFAULT_PREFIX)]
+
+            command = ' '.join([command, '--resource', resource])
+
+        try:
+            get_access_token_stdout = await self._get_cli_access_token(command)
+            get_access_token_object = json.loads(get_access_token_stdout)
+            access_token = get_access_token_object['accessToken']
+        except ClientAuthenticationError:
+            raise
+        except Exception as e:
+            raise ClientAuthenticationError(repr(e))
```

Comment:
When we land here, we know we got unexpected output from the CLI. "KeyError('accessToken')" is a poor error message for that. It's hard to give an actionable message here because we don't know what went wrong, but something like "Azure CLI didn't provide an access token" is an improvement.